src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -1007,6 +1016,22 @@ public ResponseEntity<Void> addUserToCourseGroup(String userLogin, User instruct
                 throw new EntityNotFoundException("User", userLogin);
             }
             courseService.addUserToGroup(userToAddToGroup.get(), group, role);
+
+            final List<String> channelNames = Arrays.stream(DefaultChannelType.values()).toList().stream().map(DefaultChannelType::getName).toList();
+
+            List<Course> courses = switch (role) {
+                case STUDENT -> courseRepository.findCoursesByStudentGroupName(group);
+                case TEACHING_ASSISTANT -> courseRepository.findCoursesByTeachingAssistantGroupName(group);
+                case INSTRUCTOR -> courseRepository.findCoursesByInstructorGroupName(group);
+                default -> List.of();
+            };
+
+            for (Course c : courses) {
+                channelRepository.findChannelsByCourseId(c.getId()).stream().filter(channel -> channelNames.contains(channel.getName())).forEach(channel -> {
+                    conversationService.registerUsersToConversation(c, Set.of(userToAddToGroup.get()), channel, Optional.empty());
+                });
+            }
```

Comment:
I'd move this into the corresponding service

Reply:
Moved to conversation service 👍